### DISCUSSION TITLE
Pulling questions in a single page of jsPsychSurvey from timeline variables

### DISCUSSION THREAD

**Discussion by mrinmayik** [jspsych/jsPsych#3058] [Q&A] [answered]
Hello jsPsych gurus,

I have two questions relating to jsPsychSurvey:
 - I want a single survey page with multiple questions. But I want the questions in each page to be populated from values in a timeline variable. Is that possible? Modified versions of solutions mentioned [here](https://github.com/jspsych/jsPsych/discussions/2933) and [here](https://github.com/jspsych/jsPsych/discussions/2160) will add multiple pages in each trial, with each page containing one question, right?
 - When using the "text" type in jsPsychSurvey, is it possible to have two text boxes? Similar to what one might do with jsPsychCloze, but with pages containing multiple questions.

Thank you!
Mrinmayi

**Comment by ChristopheBossens** [comment, marked as answer]
Hi mrinmayik,

For your  first question, do you mean that you want to fill in the pages property of the suvrey as a timeline variable? In that case, you could use code such as the one below to make that happen:

```js
  const jsPsych = initJsPsych();

    let timeline_questions = [
        {
            pages : [[{
                type : 'html',
                prompt : 'First page'
            },
            {
                type: 'multi-choice',
                prompt: "Which of the following do you like the most?", 
                name: 'VegetablesLike', 
                options: ['Tomato', 'Cucumber', 'Eggplant', 'Corn', 'Peas'], 
                required: true
            }, 
        ]]},
        {
            pages : [[{
                type : 'html',
                prompt : 'Second page'
            },
            {
                type: 'text',
                prompt: "Two textboxes", 
                name: 'textbox_1', 
            }, 
            {
                type: 'text',
                prompt: ' ',
                name: 'textbox_2', 
            }, 
        ]]},
    ]

    var trial = {
        type: jsPsychSurvey,
        pages: jsPsych.timelineVariable('pages')
    };

    let test_procedure = {
        timeline : [trial],
        timeline_variables : timeline_questions
    }


    jsPsych.run([test_procedure]); 
```

For your second question, a textbox question only has question. But you could create a second text question and use an empty prompt so that it looks as if you have two text boxes?

**Reply by mrinmayik** [reply]
Oh okay! It wasn't exactly what I was looking for, but it gave me the idea to build the pages variable using an external JSON file (that is output based on some experimental parameters.

Sorry - I should've thought of that. Thanks a lot for your help!!
